app/main.py

```diff
@@ -1,3 +1,28 @@
-def shop_trip():
-    # write your code here
-    pass
+from app.functions import (
+    get_dict_from_json,
+    trip,
+    costs_value,
+)
+
+
+def shop_trip() -> None:
+    data = get_dict_from_json()
+    customers = data["customers"]
+    shops = data["shops"]
+
+    for customer in customers:
+        # amount of money
+        print(f"{customer.name} has "
```

Comment:
You're using the `name` and `money` attributes of the `customer` object directly. However, you should use getters to access them. For example, use `customer.get_name()` and `customer.get_money()` instead.